docs/admin/out-of-resource.md

```diff
@@ -35,20 +39,68 @@ summary API.
 | `imagefs.available` | `imagefs.available` := `node.stats.runtime.imagefs.available` |
 | `imagefs.inodesFree` | `imagefs.inodesFree` := `node.stats.runtime.imagefs.inodesFree` |
 
-Each of the above signals supports either a literal or percentage based value.  The percentage based value
-is calculated relative to the total capacity associated with each signal.
+The value for `memory.available` is derived from the cgroupfs instead of tools
+like `free -m`.  This is important because `free -m` does not work in a
+container, and if users use the [node
+allocatable](/docs/admin/node-allocatable.md) feature, out of resource decisions
+are made local to the end user pod part of the cgroup hierarchy as well as the
+root node.  The following script simulates the same set of steps that the
+`kubelet` performs to calculate `memory.available`.  The `kubelet` excludes
+inactive_file (i.e. # of bytes of file-backed memory on inactive LRU list) from
+its calculation as it assumes that memory is reclaimable under pressure.
+
+```sh
+#!/bin/bash
+#!/usr/bin/env bash
+
+# This script reproduces what the kubelet does
+# to calculate memory.available relative to root cgroup.
+
+# current memory usage
+memory_capacity_in_kb=$(cat /proc/meminfo | grep MemTotal | awk '{print $2}')
+memory_capacity_in_bytes=$((memory_capacity_in_kb * 1024))
+memory_usage_in_bytes=$(cat /sys/fs/cgroup/memory/memory.usage_in_bytes)
+memory_total_inactive_file=$(cat /sys/fs/cgroup/memory/memory.stat | grep total_inactive_file | awk '{print $2}')
+
+memory_working_set=$memory_usage_in_bytes
+if [ "$memory_working_set" -lt "$memory_total_inactive_file" ];
+then
+    memory_working_set=0
+else
+    memory_working_set=$((memory_usage_in_bytes - memory_total_inactive_file))
+fi
+
+memory_available_in_bytes=$((memory_capacity_in_bytes - memory_working_set))
+memory_available_in_kb=$((memory_available_in_bytes / 1024))
+memory_available_in_mb=$((memory_available_in_kb / 1024))
+
+echo "memory.capacity_in_bytes $memory_capacity_in_bytes"
+echo "memory.usage_in_bytes $memory_usage_in_bytes"
+echo "memory.total_inactive_file $memory_total_inactive_file"
+echo "memory.working_set $memory_working_set"
+echo "memory.available_in_bytes $memory_available_in_bytes"
+echo "memory.available_in_kb $memory_available_in_kb"
+echo "memory.available_in_mb $memory_available_in_mb"
+```
+
+Each of the above signals supports either a literal or percentage based value.
```

Comment:
we should move this paragraph back where it was, below the signals table since we refer to "above signals"